docs/recipes/gulp/gulpfile.js

```diff
@@ -9,70 +9,71 @@
 const gulp = require('gulp');
 const connect = require('gulp-connect');
 const lighthouse = require('lighthouse');
-const ChromeLauncher = require('lighthouse/lighthouse-cli/chrome-launcher').ChromeLauncher;
+const chromeLauncher = require('lighthouse/chrome-launcher/chrome-launcher');
 const perfConfig = require('lighthouse/lighthouse-core/config/perf.json');
 const PORT = 8080;
-let chromeLauncher;
 
 /**
  * Start server
  */
 const startServer = function() {
-  return connect.server({
-    root: './public',
-    livereload: true,
-    port: PORT
-  });
+    return connect.server({
+        root: './public',
+        port: PORT
+    });
 };
 
 /**
  * Stop server
  */
 const stopServer = function() {
-  connect.serverClose();
-  chromeLauncher.kill();
-  chromeLauncher = null;
+    connect.serverClose();
 };
 
 /**
  * Run lighthouse
  */
-const runLighthouse = function() {
-  const url = `http://localhost:${PORT}/index.html`;
-  const lighthouseOptions = {}; // available options - https://github.com/GoogleChrome/lighthouse/#cli-options
-  return lighthouse(url, lighthouseOptions, perfConfig);
-};
+function launchChromeAndRunLighthouse(url, flags, config = null) {
+    return chromeLauncher.launch().then(chrome => {
+        flags.port = chrome.port;
+        return lighthouse(url, flags, config).then(results =>
+            chrome.kill().then(() => results)
+        );
+    });
+}
 
 /**
  * Handle ok result
  * @param {Object} results - Lighthouse results
  */
 const handleOk = function(results) {
-  stopServer();
-  console.log(results); // eslint-disable-line no-console
-  // TODO: use lighthouse results for checking your performance expectations.
-  // e.g. process.exit(1) or throw Error if score falls below a certain threshold.
-  return results;
+    stopServer();
+    console.log(results); // eslint-disable-line no-console
+    // TODO: use lighthouse results for checking your performance expectations.
+    // e.g. process.exit(1) or throw Error if score falls below a certain threshold.
+    // if (results.audits['first-meaningful-paint'].rawValue > 3000) {
+    //     console.log(`Warning: Time to first meaningful paint ${results.audits['first-meaningful-paint'].displayValue}`);
+    //     process.exit(1);
+    // }
+    return results;
 };
 
 /**
  * Handle error
  */
 const handleError = function(e) {
-  stopServer();
-  console.error(e); // eslint-disable-line no-console
-  throw e; // Throw to exit process with status 1.
+    stopServer();
+    console.error(e); // eslint-disable-line no-console
+    throw e; // Throw to exit process with status 1.
 };
 
-gulp.task('lighthouse', function() {
-  chromeLauncher = new ChromeLauncher();
+const flags = { output: 'json' };
```

Comment:
`{output: 'json'}`. But FWIW, the [module doesn't respect the `output` flag](https://github.com/GoogleChrome/lighthouse/issues/2401). That's only a CLI option atm. Probably should just remove so there's no confusion.

Can we also keep the comment `// available options - https://github.com/GoogleChrome/lighthouse/#cli-options`?